src/Servers/Kestrel/Core/src/Internal/Http/HttpParser.cs

```diff
@@ -415,9 +416,29 @@ private unsafe Span<byte> GetUnknownMethod(byte* data, int length, out int metho
             return new Span<byte>(data, methodLength);
         }
 
+        private unsafe bool IsTlsHandshake(byte* data, int length)
+        {
+            const byte SslRecordTypeHandshake = (byte)0x16;
+
+            //make sure we can check at least for the existence of a TLS handshake - we check the first three bytes
```

Comment:
This comment says 3 bytes but you only check one?

Reply:
Thank you!

Initially I checked for three bytes - then I changed my solution to just the check for 0x16 after @davidfowl's comment. Shall I correct the comment and/or the message?

Reply:
Yes please.

Reply:
Will commit ASAP!